eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -297,6 +298,37 @@ public function loadLocationChildren(APILocation $location, $offset = 0, $limit
         );
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function loadParentLocationsForDraftContent(VersionInfo $versionInfo)
+    {
+        if (!$versionInfo->isDraft()) {
+            throw new BadStateException(
+                '$contentInfo',
+                sprintf(
+                    'Content [%d] %s has been already published. Use LocationService::loadLocations instead.',
+                    $versionInfo->contentInfo->id,
+                    $versionInfo->contentInfo->name
+                )
+            );
+        }
+        $spiLocations = $this->persistenceHandler
+            ->locationHandler()
+            ->loadParentLocationsForDraftContent($versionInfo->contentInfo->id);
+
+        $locations = [];
+        $permissionResolver = $this->repository->getPermissionResolver();
+        foreach ($spiLocations as $spiLocation) {
+            $location = $this->domainMapper->buildLocationDomainObject($spiLocation);
+            if ($permissionResolver->canUser('content', 'read', $location->contentInfo, [$location])) {
```

Comment:
`->getContentInfo()`, skip the magic :)

Reply:
Magic is gone, visiting Hogwarts, via a803f6a :laughing: